darwin/Platform.c

```diff
@@ -120,21 +120,65 @@ const MeterClass* const Platform_meterTypes[] = {
    NULL
 };
 
-static double Platform_nanosecondsPerMachTick = 1.0;
+static void Platform_getCPUBrandString(char *cpu_brand_string, size_t cpu_brand_string_size) {
+   if (sysctlbyname("machdep.cpu.brand_string", cpu_brand_string, &cpu_brand_string_size, NULL, 0) == -1) {
+      CRT_fatalError("Could not determine what CPU we are running on");
+   }
+}
 
-static double Platform_schedulerTicksPerNS = -1;
+// Adapted from https://developer.apple.com/documentation/apple-silicon/about-the-rosetta-translation-environment
+static bool Platform_isRunningUnderRosetta2() {
+   int ret = 0;
+   size_t size = sizeof(ret);
+   if (sysctlbyname("sysctl.proc_translated", &ret, &size, NULL, 0) == -1) {
+      if (errno == ENOENT) return 0;
+      CRT_fatalError("Could not determine if this process was running in a translation environment like Rosetta 2.");
```

Comment:
Is this really fatal?  Maybe issue a warning here, and return 0 as with ENOENT... not clear what other conditions this might happen under.

Reply:
> Maybe issue a warning here

And how do I do that? 😄 

> not clear what other conditions this might happen under.

Me neither. In Apple's provided code, they returned a `-1` (their return type was `int`, not `bool` like here). I think it's probably safe to assume that if there's an error here, we'll assume we're *not* running under Rosetta.

Reply:
> > Maybe issue a warning here
> 
> And how do I do that?

Just using "fprintf(stderr, ..." is the only option I believe, and falling back to existing behaviour.  When the user exits they'll see the warning and can scratch their head accordingly, at least slightly better informed and with some clues.

> probably safe to assume that if there's an error here, we'll assume we're _not_ running under Rosetta.

+1

Reply:
Done:

https://github.com/amomchilov/htop/blob/737dd931b7ed69bbf6b8a68986a18fb7c8a0ffd8/darwin/PlatformHelpers.c#L75-L83